code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -666,3 +666,25 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 /atom/proc/create_reagents(const/max_vol)
 	reagents = new/datum/reagents(max_vol)
 	reagents.my_atom = src
+
+///Used by grinders////
+/datum/reagents/proc/grind_item(var/obj/item/I, var/process = 1)
```

Comment:
Your implementation seems really clunky to execute because it's a line right in the middle of a block of code that you want to override. Whether to tell it to record data, change the amount, tell it to calculate an amount... it's right in the middle of a block of other code you want to keep.